### PR TITLE
extract spiffe id format into pki util.

### PR DESCRIPTION
**Pull request by incfly**

### CODE REVIEW

File: security/pkg/pki/util/generate_cert.go [hunk 1]
```diff
@@ -135,6 +135,11 @@ func LoadSignerCredsFromFiles(signerCertFile string, signerPrivFile string) (*x5
 	return cert, key, nil
 }
 
+// GenSubjectName returns the formatted subject name for the certificate (SPIFFEE format for now).
+func GenSubjectName(ns, serviceAccount string) string {
```
Comment (myidpt) [inline security/pkg/pki/util/generate_cert.go]: GenSanUriName? As this is not the subject field in the certs.
Then maybe we can put this function in san.go:
https://github.com/istio/istio/blob/master/security/pkg/pki/util/san.go

File: security/cmd/node_agent/management/server.go [hunk 2]
```diff
@@ -150,12 +150,9 @@ func (s *Server) WorkloadAdded(ctx context.Context, request *pb.WorkloadInfo) (*
 		return fmt.Errorf(msg)
 	}
 
-	// Sends CSR request to CA.
-	// TODO(inclfy): extract the SPIFFE formatting out into somewhere else.
-	id := fmt.Sprintf("spiffe://cluster.local/ns/%s/sa/%s", ns, sa)
-
+	// Sends CSR request to CA. .
```
Comment (wattli) [inline security/cmd/node_agent/management/server.go]: Remove extra .

File: security/pkg/pki/util/san_test.go [hunk 3]
```diff
@@ -227,3 +227,10 @@ func TestExtractIDs(t *testing.T) {
 		}
 	}
 }
+
+func TestGenSanURI(t *testing.T) {
+	got := GenSanURI("namespace-bar", "service-foo")
+	if expected := "spiffe://cluster.local/ns/namespace-bar/sa/service-foo"; got != expected {
```
Comment (wattli) [inline security/pkg/pki/util/san_test.go]: Also add a negative test?

**Comment by incfly** [comment]
@myidpt @wattli could some of you stamp lgtm again? thx!

**Pull request merged**